### DISCUSSION TITLE
Memory usage is increasing when scrolling in ListBox

### DISCUSSION THREAD

**Discussion by HaythemJ** [AvaloniaUI/Avalonia#15457] [Q&A]
Hi,

I am developping an application where I am using a ListBox displaying products with theirs images,prices, qunatity ...

Everything is working fine but I am getting a strange behavior for the memory usage. Whenever I am scrolling up and down in the list the memory is getting increasing. I tried to analyse the memory usage and I found that a huge number of new objects are created what explains the increase of the memory usage:

![image](https://github.com/AvaloniaUI/Avalonia/assets/9905290/09d99e35-17ab-4ba2-b40a-f5eb3683d244)

From Snapshot 2 I just scrolled up and down

From Snapshot 4 I just scrolled up and down

From Snapshot 5 I did nothing and I kept the application running and after a while I took a snapshot

For the images loading I am using [[AsyncImageLoader.Avalonia](https://github.com/AvaloniaUtils/AsyncImageLoader.Avalonia)] with a cache mechanism. 

I investigated my own objects and they are stable.

I am wondering if I am doing it the right way.

Here is my XAML code. 

![image](https://github.com/AvaloniaUI/Avalonia/assets/9905290/65c816e6-e88b-42b6-be86-ee9cc061fa40)

**Comment by stevemonaco** [comment]
I'd suggest using a memory profiler that allows you to at least see the type of what's allocated. VS has Performance Profiler -> .NET Object Allocation Tracking. dotMemory has similar.

I'd suspect the SVG `Image`, but it's a bit odd that usage went up a bit while idle.

Besides that, `ListBox` already has its own `ScrollViewer` and uses `VirtualizingStackPanel` by default. Those shouldn't be related to the problem though.

**Comment by timunie** [comment]
I'd suggest to not add a ScrollViewer around the ListBox. It has one on it's own. An outer one could break virtualization.

**Comment by thevortexcloud** [comment]
This has been discussed in depth before. Nobody seemed to know why it happens. However I ended up working around it by adding my own code to dispose of the image when needed and to also handle scaling the images myself. My memory usage now rarely goes about ~300mb when displaying >2k images.

https://github.com/AvaloniaUI/Avalonia/issues/14304
